client/castApp/src/dbcb.c

```diff
@@ -58,14 +62,23 @@ static int pushEnv(caster_t *caster)
     if(ret)
         ERRRET(ret, caster, "Failed to send epics version");
 
-    for(i=0; !ret && envs[i]; i++) {
-        const char *val = getenv(envs[i]);
+    for(i=0; !ret && default_envs[i]; i++) {
+        const char *val = getenv(default_envs[i]);
         if(val && val[0]!='\0')
-            ret = casterSendInfo(caster, 0, envs[i], val);
+            ret = casterSendInfo(caster, 0, default_envs[i], val);
         if(ret)
-            casterMsg(caster, "Error sending env %s", envs[i]);
+            casterMsg(caster, "Error sending env %s", default_envs[i]);
     }
 
+    if(caster->extra_envs) {
+        for (i = 0; !ret && caster->extra_envs[i]; i++) {
```

Comment:
I think if one removes the NULL-termination of the `extra_envs` variable this can likely become simpler; we don't even need the `if (caster->extra_envs)` for example.

Reply:
Need some locking for accesses of and through `extra_envs[]`.

Reply:
Resolved both comments w/ https://github.com/ChannelFinder/recsync/pull/57/commits/908052b7244a1ed8b477216a6f0cc3a41d5b3176